pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1151,34 +866,32 @@ protected IdealState ensureAllPartitionsConsuming(final TableConfig tableConfig,
           // 1. all replicas OFFLINE and metadata IN_PROGRESS/DONE - a segment marked itself OFFLINE during consumption for some reason
           // 2. all replicas ONLINE and metadata DONE - Resolved in https://github.com/linkedin/pinot/pull/2890
           // 3. we should never end up with some replicas ONLINE and some OFFLINE.
-          if (isAllInstancesInState(instanceStateMap, PinotHelixSegmentOnlineOfflineStateModelGenerator.OFFLINE_STATE)
-              || !(isTooSoonToCorrect(tableNameWithType, segmentId, now))) {
-
-            // No instances are consuming, so create a new consuming segment.
-            LLCSegmentName newLLCSegmentName = makeNextLLCSegmentName(segmentName, partition, now);
-            LOGGER.info("Creating CONSUMING segment {} for {} partition {}", newLLCSegmentName.getSegmentName(),
-                tableNameWithType, partition);
-
-            // To begin with, set startOffset to the oldest available offset in the stream. Fix it to be the one we want,
-            // depending on what the prev segment had.
-            long startOffset = getPartitionOffset(streamConfig, OffsetCriteria.SMALLEST_OFFSET_CRITERIA, partition);
-            LOGGER.info("Found smallest offset {} for table {} for partition {}", startOffset, tableNameWithType,
-                partition);
-            startOffset = getBetterStartOffsetIfNeeded(tableNameWithType, partition, segmentName, startOffset,
-                newLLCSegmentName.getSequenceNumber());
+          if (isAllInstancesInState(instanceStateMap, RealtimeSegmentOnlineOfflineStateModel.OFFLINE)) {
```

Comment:
You need to also check if it is too soon to correct this segment.

Reply:
You are not handling the case where all instances are in ONLINE state, but there are no CONSUMING segments. Basically, handle PR #2891

Reply:
We don't need to check for all OFFLINE case. That is not a valid state for segment commit.
The `all in ONLINE state` scenario is handled in RetentionManager (#2890 as in the comments). #2891 does not handle that scenario as well.